SalesforceCommerceCloud/pwa-kit
# @W-14810956@ Update generator with options for slas private client

This change updates the pwa-kit-create-app project to include an option for setting up a SLAS private client.

Preset projects have been set up to default use a SLAS public client.

A new preset `retail-react-app-private-slas-client` has been added to the generator. This preset will set up an extensible project against the ODS `zzrf-001`. Ask @vcua-mobify for the client secret associated with using this preset

Testing the changes:
1. Check out the changes in this PR
2. Generate a project (you can either use the preset `retail-react-app-private-slas-client` or enter values manually)
3. Make sure you set the `PWA_KIT_SLAS_CLIENT_SECRET` environment variable to your client secret
4. Build and start the generated project
5. See that the project starts and is using the SLAS private client flow (no calls to SLAS /authorize)

The command for using the preset locally is below:
`./scripts/create-mobify-app-dev.js --preset=retail-react-app-private-slas-client`


Note: These changes apply to both an extensible project and a non-extensible project. Changes have been tested to work with both.
